### PR TITLE
bug[ASM-10942]- Required: add option to not enforce required fields

### PR DESCRIPTION
**Pull request by dan-akeyless**

### CODE REVIEW

**Review by alikdolg — APPROVED** [review]

File: cli_test.go [hunk 1]
```diff
@@ -58,8 +58,10 @@ type argT struct {
 	Float32 float32 `cli:"f32" usage:"type float32"`
 	Float64 float64 `cli:"f64" usage:"type float64"`
 
-	PtrBool *bool `cli:"ptrBool"`
-	PtrStr *string `cli:"ptrStr"`
+	PtrBool *bool   `cli:"ptrBool"`
+	PtrStr  *string `cli:"ptrStr"`
+
+	RequiredNotEnforced *string `cli:"*requiredNotEnforced" ignoreRequired:"true"`
```
Comment (tuvia-akeyless) [inline cli_test.go]: where is it tested?

Comment (dan-akeyless) [reply cli_test.go]: in all the subtests, they would have needed `--requiredNotEnforced` otherwise

File: tag.go [hunk 2]
```diff
@@ -101,8 +103,11 @@ func parseTag(fieldName string, structTag reflect.StructTag) (p *tagProperty, is
 	p.dft = tag.Get(tagDefaut)
 
 	// `ignoreDft` TAG
-	toIgnore, _ := strconv.ParseBool(tag.Get(tagDefautInject))
-	p.ignoreDft = toIgnore
+	dftToIgnore, _ := strconv.ParseBool(tag.Get(tagIgnoreDefault))
+	p.ignoreDft = dftToIgnore
+
+	requiredToIgnore, _ := strconv.ParseBool(tag.Get(tagIgnoreRequired))
```
Comment (tuvia-akeyless) [inline tag.go]: where is the error checking?

File: tag_test.go [hunk 3]
```diff
@@ -110,7 +112,7 @@ func TestTag(t *testing.T) {
 			assert.False(t, tag.isEdit)
 			assert.Equal(t, tag.longNames, []string{"--sep"})
 			assert.Equal(t, tag.sep, ":")
-		case "Required":
+		case "Required", "RequiredNotEnforced":
```
Comment (tuvia-akeyless) [inline tag_test.go:115]: what about unittest for RequiredNotEnforced specifically?

File: cli_test.go (hunk 1 above)
Comment (tuvia-akeyless) [reply cli_test.go]: Right. Can you add subtest with this flag?

File: tag_test.go (hunk 3 above)
Comment (dan-akeyless) [reply tag_test.go:115]: This is the place

File: tag.go (hunk 2 above)
Comment (dan-akeyless) [reply tag.go]: if someone puts something other than "1", "t", "T", "true", "TRUE", "True", "0", "f", "F", "false", "FALSE", "False" for example "" then we'll assume it's false

File: cli_test.go (hunk 1 above)
Comment (dan-akeyless) [reply cli_test.go]: sure

File: tag_test.go (hunk 3 above)
Comment (tuvia-akeyless) [reply tag_test.go:115]: I mean different unittests for enforced and not enforced that check the results are different in the relevant area

Comment (dan-akeyless) [reply tag_test.go:115]: No the idea of this test is that the output in --help for Required and RequiredNotEnforced is the same
There is no logic that is run here.
The logic tests are run above

File: tag.go (hunk 2 above)
Comment (tuvia-akeyless) [reply tag.go]: consider typo in the tag - should return error

**Review by tuvia-akeyless — APPROVED** [review]

File: tag.go (hunk 2 above)
Comment (dan-akeyless) [reply tag.go]: done

**Pull request merged**